epub3-to-daisy202/src/main/resources/xml/xslt/augment-smil.xsl

```diff
@@ -0,0 +1,114 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0"
+                xmlns:html="http://www.w3.org/1999/xhtml"
+                xmlns:xs="http://www.w3.org/2001/XMLSchema"
+                xmlns:pf="http://www.daisy.org/ns/pipeline/functions"
+                xmlns="" xpath-default-namespace=""
+                exclude-result-prefixes="#all">
+    
+    <xsl:import href="http://www.daisy.org/pipeline/modules/file-utils/library.xsl"/>
+    
+    <!--
+        the smil
+    -->
+    <xsl:variable name="smil" select="collection()[1]/*"/>
+    <!--
+        the corresponding content document(s) in spine order
+    -->
+    <xsl:variable name="html" select="collection()[position()&gt;1]/*"/>
+    
+    <xsl:variable name="smil-base-uri" select="base-uri($smil)"/>
+    
+    <xsl:key name="absolute-id" match="*[@id]" use="concat(base-uri(root()/*),'#',@id)"/>
```

Comment:
OK good point. Perhaps a helper function in html-utils to deal with this?

Reply:
By the way now I am not even sure anymore that the `base-uri(root()/*)` actually works, because of the Saxon/Calabash issue mentioned below. It does work in the test, but possibly not in real situations because the base URI is set differently.